portal/server/helpers/getBusinessDayOfMonth.js

```diff
@@ -0,0 +1,79 @@
+const { addBusinessDays, isSameDay, isWeekend, startOfMonth } = require('date-fns');
+
+/**
+ * @param {Date} date
+ * @param {Date[]} holidays
+ * @returns {boolean}
+ */
+const isHoliday = (date, holidays) => holidays.some((holiday) => isSameDay(date, holiday));
+
+/**
+ * @param {Date} date
+ * @param {Date[]} holidays
+ * @returns {Date}
+ */
+const addOneBusinessDayWithHolidays = (date, holidays) => {
+  let nextBusinessDay = addBusinessDays(date, 1);
+  while (isHoliday(nextBusinessDay, holidays)) {
+    nextBusinessDay = addBusinessDays(nextBusinessDay, 1);
+  }
+  return nextBusinessDay;
+};
+
+/**
+ * Given a month, year and an array of holidays, this returns
+ * the first business day of the month in the year provided.
+ * For example, if the 1st of the month is a Monday and that
+ * day is not a holiday, it will return the current date. If
+ * the 1st is a Saturday, and the next Monday is not a holiday,
+ * it will return the next Monday.
+ * @param {number} month - The month (one-indexed)
+ * @param {number} year - The year
+ * @param {Date[]} holidays - A list of dates which should be excluded as holidays
+ * @returns {Date}
+ */
+const getFirstBusinessDayOfMonth = (month, year, holidays) => {
+  let firstBusinessDay = startOfMonth(new Date(year, month - 1));
+  if (isWeekend(firstBusinessDay)) {
+    firstBusinessDay = addOneBusinessDayWithHolidays(firstBusinessDay, holidays);
+  }
+  while (isHoliday(firstBusinessDay, holidays)) {
+    firstBusinessDay = addOneBusinessDayWithHolidays(firstBusinessDay, holidays);
+  }
+  return firstBusinessDay;
+};
+
+/**
+ * Given a month (and year), an array of holidays and an
+ * index, it will return the business day with that index.
+ * If the index is one, and the start date is a business
+ * day, it will return the date provided as that is the
+ * first-indexed business day.
+ * @param {number} month - The month (one-indexed)
+ * @param {number} year - The year
+ * @param {Date[]} holidays - A list of dates which should be excluded as holidays
+ * @param {number} businessDay - The one-indexed business day to get
+ * @returns {Date}
+ */
+const getBusinessDayOfMonth = (month, year, holidays, businessDay) => {
```

Comment:
I think this be a little cleaner if instead of `month` and `year` we just passed in a Date object which represents a date in the relevant month (maybe called something like `dateInMonth`. Then where this is called we don't need to worry about passing the values in separately, and also we don't need to remember that the month needs to be one-indexed. That logic can instead all be taken care of here.